Source/Lib/TIFF/TIFF.cpp

```diff
@@ -0,0 +1,336 @@
+/*  Copyright (c) MediaArea.net SARL & AV Preservation by reto.ch.
+ *
+ *  Use of this source code is governed by a BSD-style license that can
+ *  be found in the License.html file in the root of the source tree.
+ */
+
+//---------------------------------------------------------------------------
+#include "Lib/TIFF/TIFF.h"
+#include "Lib/RAWcooked/RAWcooked.h"
+#include <sstream>
+#include <ios>
+using namespace std;
+//---------------------------------------------------------------------------
+
+//---------------------------------------------------------------------------
+// Tested cases
+struct tiff_tested
+{
+    tiff::encoding              Encoding;
+    tiff::descriptor            Descriptor;
+    uint8_t                     BitDepth;
+    tiff::packing               Packing;
+    tiff::endianess             Endianess;
+    tiff::flavor                Flavor;
+};
+
+const size_t TIFF_Tested_Size = 1;
+struct tiff_tested TIFF_Tested[TIFF_Tested_Size] =
+{
+    { tiff::Raw  , tiff::RGB      ,  8, tiff::SingleStrip   , tiff::LE, tiff::Raw_RGB_8                 },
+};
+
+//---------------------------------------------------------------------------
+
+//***************************************************************************
+// TIFF
+//***************************************************************************
+
+//---------------------------------------------------------------------------
+tiff::tiff() :
+    input_base_uncompressed(Parser_TIFF, true),
+    FrameRate(NULL)
+{
+}
+
+//---------------------------------------------------------------------------
+bool tiff::Parse(bool AcceptTruncated, bool FullCheck)
+{
+    if (Buffer_Size < 8)
+        return false;
+
+    Buffer_Offset = 0;
+    uint32_t Magic = Get_B4();
+    switch (Magic)
+    {
+        case 0x49492A00: // II*
+            IsBigEndian = false;
+            break;
+        case 0x4D4D002A: // MM *

```

Comment:
_Stricto sensu_ it’s a NUL not a space, however for readability I would suggest to use `"MM *"`, and just above `"II* "`, with quotation marks and spaces.

[_Sensu lato_ the person who used NUL values in a magic number should be severely punished.]

Reply:
Actually, the 2A00/002A thing is a number ("Answer to the Ultimate Question of Life, the Universe, and Everything"), expected to be a "version number" (if I understand well the origins of TIFF) despite the fact it is always the same number in any version up to now ("An arbitrary but carefully chosen number that further identifies the file as a TIFF file" in TIFF6).
Changed to `"II" + 42 in 16-bit hex` (and `"MM" + 42 in 16-bit hex`  respectively).

More seriously, by the way I added TIFF RGB/RGBA 8/16-bit support + AIFF/AIFC support (PCM only, Signed/Unsigned, LE/BE, 8/16/24-bit, 1/2/6-channel), still need to have more checks e.g. rejecting when pixel strips are not in a consecutive order (it is doable to support such file, but longer development and sponsors did not provide such file up to now).

Reply:
>"An arbitrary but carefully chosen number that further identifies the file as a TIFF file"

He he! I presume «carefully chosen» means 4 + 2 = 6
;-)

>Changed to "II" + 42 in 16-bit hex (and "MM" + 42 in 16-bit hex respectively).

That’s much better, for code readability. Thank you!